src/middleware/router.js

```diff
@@ -329,22 +330,27 @@ const buildNonPrimarySendRequestPromise = (ctx, route, options, path) =>
         bodyId: ctx.request.bodyId,
         timestamp: ctx.requestTimestamp
       }
-      if (response.headers != null && response.headers['content-type'] != null && response.headers['content-type'].indexOf('application/json+openhim') > -1) {
-        // handle mediator reponse
-        let payload = ''
-        response.body.on('data', (data) => {
-          payload += data.toString()
-        })
 
-        response.body.on('end', () => {
-          const responseObj = JSON.parse(payload)
+      if (response.headers != null && response.headers['content-type'] != null && response.headers['content-type'].indexOf('application/json+openhim') > -1) {
+        // handle mediator response
+        const responseObj = JSON.parse(response.body)
+        
+        if (responseObj['x-mediator-urn']) {
           routeObj.mediatorURN = responseObj['x-mediator-urn']
+        }
+        if (responseObj.orchestrations) {
           routeObj.orchestrations = responseObj.orchestrations
+        }
+        if (responseObj.properties) {
           routeObj.properties = responseObj.properties
-          if (responseObj.metrics) { routeObj.metrics = responseObj.metrics }
-          if (responseObj.error) { routeObj.error = responseObj.error }
-        })
-        routeObj.response = responseObj.response
+        }
+        if (responseObj.metrics) {
+          routeObj.metrics = responseObj.metrics
+        }
+        if (responseObj.error) {
+          routeObj.error = responseObj.error
+        }
```

Comment:
Lets maybe update these multiple if statements into a bit more cleaner way by using a ternary operator
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator
e,g `routeObj.mediatorURN = responseObj['x-mediator-urn'] ? responseObj['x-mediator-urn'] : undefined`

If it doesnt exist, it just sets the property to undefined which it would be if its not set and queried

Reply:
fixed in https://github.com/jembi/openhim-core-js/pull/1060/commits/80469141b8a989a1a554d41119ce1c757b84cb3d